ntrans_gui.py

```diff
@@ -173,18 +182,46 @@ def __init__(self) -> None:
         get_savepath_button.grid(sticky="W", column=1, row=13, padx=(0, 20), pady=(0, 0))
 
         generate_dictionary.grid(column=0, columnspan=2, padx=(0, 0), pady=(0, 0))
+        self.setting_not_defined_warning.grid(column=0, row=15, columnspan=2, padx=(0, 0), pady=(0, 0))
         self.estimated_time_label.grid(sticky="W", column=0, columnspan=2, padx=(30, 0), pady=(0, 0))
 
         about_help_buttonframe.grid(sticky="E", column=0, row=17, columnspan=2, padx=(20, 20), pady=(20, 10))
 
         # Black turn on formatting
         # fmt: on
 
-        self.select_all_var.trace_add("write", self.update_estimated_time_label)
-        self.data_size_var.trace_add("write", self.update_estimated_time_label)
+        self.filepath.trace_add("write", self.update_user_choice_vars)
+        self.target_language_var.trace_add("write", self.update_user_choice_vars)
         for var in self.checkbox_vars.values():
+            var.trace_add("write", self.update_user_choice_vars)
             var.trace_add("write", self.update_estimated_time_label)
 
+        self.select_all_var.trace_add(
+            "write", self.update_estimated_time_label
+        )  # TODO: Is this trace needed?
+        self.data_size_var.trace_add("write", self.update_user_choice_vars)
+
+        self.select_all_var.trace_add("write", self.update_estimated_time_label)
+        self.data_size_var.trace_add("write", self.update_estimated_time_label)
+
+        self.update_estimated_time_label()
+
+    def flash_warning_label(self) -> None:
+        # print("In flashing method")
+        # self.style_options.configure("W.TLabel", foreground="white")
+        # self.style_options.configure("W.TLabel", foreground="red")
+        pass  # TODO: style.configure doesn't work.
```

Comment:
Arrinao's tetris has flashing code that works pretty well. You can ctrl+f flash in it.